src/main/java/uk/org/tombolo/importer/DownloadUtils.java

```diff
@@ -42,13 +46,31 @@ public File fetchFile(URL url, String prefix, String suffix) throws IOException{
 	public InputStream fetchInputStream(URL url, String prefix, String suffix) throws IOException {
 		createCacheDir(prefix);
 		File localDatasourceFile = urlToLocalFile(url, prefix, suffix);
+
+
 		log.info("Fetching local file: {}", localDatasourceFile.getCanonicalPath());
 		if (!localDatasourceFile.exists()){
 			log.info("Local file not found: {} \nDownloading external resource: {}",
 												localDatasourceFile.getCanonicalPath(), url.toString());
-			URLConnection connection = url.openConnection();
-			if (suffix.equals(".json")) connection.setRequestProperty("Accept", "application/json");
-			return new TeeInputStream(connection.getInputStream(), new FileOutputStream(localDatasourceFile));
+
+			URLConnection urlConnection = url.openConnection();
```

Comment:
Rather casting the ```URLConnection``` object later to ```HttpURLConnection``` and checking ```instanceOf``` we could here just say ```HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();```

Reply:
This would eliminate the need to checking ```instanceOf``` and type casting

Reply:
What about the elusive `JarURLConnection`